File: implement-shell-tools/cat/catfile.js
const fs = require("fs");

const args = process.argv.slice(2);

const showNumber = args.includes("-n");

const files = args.filter((arg) => !arg.startsWith("-"));

files.forEach((file) => {
  const content = fs.readFileSync(file, "utf-8");
  const lines = content.split("\n");

  lines.forEach((line, index) => {
    if (showNumber) {
      console.log(`${index + 1} ${line}`);
    } else {
      console.log(line);
    }
  });
});

File: implement-shell-tools/cat/catfile.py
import sys

# Get command line arguments (skip script name)
args = sys.argv[1:]

# Check if -n flag exists
show_number = "-n" in args

# Get file names (ignore flags)
files = [arg for arg in args if not arg.startswith("-")]

for file in files:
    with open(file, "r") as f:
        lines = f.read().split("\n")

        for index, line in enumerate(lines):
            if show_number:
                print(f"{index + 1} {line}")
            else:
                print(line)